app/src/main/java/org/oppia/android/app/mydownloads/MyDownloadsActivity.kt

```diff
@@ -17,20 +19,30 @@ class MyDownloadsActivity : InjectableAppCompatActivity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     activityComponent.inject(this)
-    myDownloadsActivityPresenter.handleOnCreate()
+    val isFromNavigationDrawer = intent.getBooleanExtra(
+      BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY,
+      /* defaultValue= */ false
+    )
     internalProfileId = intent.getIntExtra(KEY_NAVIGATION_PROFILE_ID, -1)
+    myDownloadsActivityPresenter.handleOnCreate(internalProfileId, isFromNavigationDrawer)
+    title = getString(R.string.menu_my_downloads)
   }
 
   companion object {
-    fun createMyDownloadsActivityIntent(context: Context, profileId: Int?): Intent {
+
+    const val BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY =
```

Comment:
Why do we need this boolean?
This is required for Help and Options because we can access then from Navgation Drawer as well as Exploration but for this it is not required as this is only accessible from Navigation Drawer and therefore always true.

Reply:
Keeping this open as of now, looking into the edge case if there is any chance of coming to MyDownloadsScreen from TopicInfo.

Reply:
Okay, do let me know once this is clear as it will change code in multiple files.